### ISSUE TITLE
Feature request: more completions!

### ISSUE THREAD

**Issue by jyn514** [chriswalz/bit#23]
Some commands I use often that aren't currently suggested (or need 90% of the typing to be suggested):

- `git rebase -i $(git merge-base HEAD master)` - this one is a stretch
- `git rebase -i master` - bit will not complete `master`, you have to type it in yourself
- `git status` - for some reason typing `st` shows `pull --rebase origin master` before commands starting with st?
- `git show` - same bug, shows commads that don't start with `sh` before show
- `git checkout <branch> -- file` - bit will complete up to <branch> but not further
- `git checkout -p`; `git add -p` - bit will not show `-p` in the initial list. This one is a stretch.
- `git log <branch>` - bit will not complete `<branch>`
- `git diff --cached` - `diff` needs up to `dif` to complete and `--cached` isn't suggested at all
- `git worktree add` - bit will complete up to `worktree` but no further

**Comment by chriswalz** [comment]
bit 39d807e24c5be4fc081e29623ed09d0dd13d497c will now:

- searches for suggestions based on prefix except for branches where it will do a "contains" filter
- shows branches for log & rebase


Future work: 
git checkout <branch> -- file - bit will complete up to <branch> but not further
git checkout -p; git add -p - bit will not show -p in the initial list. This one is a stretch.
git worktree add - bit will complete up to worktree but no further
